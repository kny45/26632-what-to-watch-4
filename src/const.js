export const MovieScore = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERRY_GOOD: 10
};

export const MovieLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERRY_GOOD: `Verry good`,
  AWESOME: `Awesome`
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const GENRE_ALL = `All genres`;

export const SHOWED_ITEMS_IN_MOVIES_LIST_DEFAULT = 8;

export const TIME_OUT = 1000;

export const VideoPlayerMode = {
  SMALL_MOVIE_CARD: `small movie card`,
  BIG_MOVIE_PLAYER: `big movie player`,
};
