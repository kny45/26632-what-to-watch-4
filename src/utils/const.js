export const MovieScore = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERRY_GOOD: 10
};

export const MovieLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERRY_GOOD: `Verry good`,
  AWESOME: `Awesome`
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const GENRE_ALL = `All genres`;

export const SHOWED_ITEMS_IN_MOVIES_LIST_DEFAULT = 8;

export const TIME_OUT = 1000;

export const VideoPlayerMode = {
  SMALL_MOVIE_CARD: `small movie card`,
  BIG_MOVIE_PLAYER: `big movie player`,
};

export const AppPage = {
  MAIN_PAGE: `main screen`,
  MOVIE_PAGE: `movie screen`,
  BIG_MOVIE_PLAYER: `big movie player`,
  SIGN_IN: `sign in screen`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  SING_IN: `/sign-in`,
  MY_LIST: `/mylist`,
  FILM: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
};
