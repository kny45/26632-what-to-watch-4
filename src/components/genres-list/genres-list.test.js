import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list.jsx";

import {GENRE_ALL} from "../../utils/const.js";

describe(`Render GenresList`, () => {
  it(`Should GenresList render correctly`, () => {
    const moviesList = [
      {
        genre: `Comedies`,
      },
      {
        genre: `Crime`,
      },
      {
        genre: `Documentary`,
      },
      {
        genre: `Drama`,
      },
      {
        genre: `Horror`,
      },
      {
        genre: `Kids & Family`,
      },
      {
        genre: `Romance`,
      },
      {
        genre: `Sci-Fi`,
      },
      {
        genre: `Thrillers`,
      },
      {
        genre: `Drama`,
      }
    ];

    const tree = renderer
      .create(
          <GenresList
            moviesList = {moviesList}
            currentGenre = {GENRE_ALL}
            onGenreItemClick = {() => {}}
          />)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
