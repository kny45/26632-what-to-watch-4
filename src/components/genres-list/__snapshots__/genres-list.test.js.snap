// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenresList Should GenresList render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Kids & Family
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Thrillers
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
</ul>
`;
